src/detection/Peak.cc

```diff
@@ -185,9 +141,7 @@ PTR(PeakTable) PeakTable::make(
         }
     }
     // No match: we create a new table and put it in the cache
-    PTR(PeakTable) newTable = std::make_shared<PeakTableImpl>(
-        schema, afw::table::IdFactory::makeSimple()
-    );
+    std::shared_ptr<PeakTable> newTable(new PeakTable(schema, afw::table::IdFactory::makeSimple()));
```

Comment:
`make_shared` needs to call a class constructor, forwarding the arguments you pass to it.  It that constructor is private or protected, `make_shared` can't call it.

I avoided that in the past here by making a hidden subclass (in an anoynmous namespace) with a public constructor that delegated to the protected one on its base class. But that hidden subclass confused pybind11's automatic downcasting abiliity, because when it used `typeid` on an instance, it doesn't return something in pybind11's registry of known types to try to cast to.